omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb

```diff
@@ -0,0 +1,322 @@
+class PostgresqlPreflightValidator < PreflightValidator
```

Comment:
This class has a lot of conditionals, early returns, special cases, etc.  Could we write some rspec tests for it where we mock out EcPostgresql?